### DISCUSSION TITLE
Read only device

### DISCUSSION THREAD

**Discussion by sauvant** [thelsing/knx#232] [Q&A] [answered]
Hi experts,

I need the ability to only read some group addresses in a KNX TP environment. I want to use an ESP32 and a Siemens 5WG1 117-2AB12 bus coupling unit to achieve that. I am struggling with the complexity of this library at the moment. Especially when it comes to integrate that with ETS.

Quick question: Do I need this integration for read only at all? Do I need a knxprod-file in this case?

Thank you and best regards!
Keith

**Comment by thelsing** [comment, marked as answer]
If you just want to read some GAs this lib is overkill. This lib is for implementing whole knx devices. 
For your use case https://github.com/franckmarini/KnxDevice or https://github.com/thorsten-gehrig/arduino-tpuart-knx-user-forum should work.

**Reply by sauvant** [reply]
Thank you for the quick answer 👍 Will have a look at these libs. I am not sure if they are ESP32 compatible though 🤔

**Reply by sauvant** [reply]
At least the latter one has a pull request for compatibility 👍

**Reply by sauvant** [reply]
Thank you Thorsten for the hint and for implementing the library! 
Worked like a charm on my environment after applying https://github.com/thorsten-gehrig/arduino-tpuart-knx-user-forum/pull/1

Best regards
Keith
